### ISSUE TITLE
Add IRIG-B Reader

### ISSUE THREAD

**Issue by stefco** [stefco/geco_data#17]
![three_irig_signals_16k_annotated](https://cloud.githubusercontent.com/assets/9324499/14550964/9e53088a-029b-11e6-8dd8-5a3928c3317f.png)

![one second of irig-b signal hanford ex from sep 14 2015 09_50_44 to sep 14 2015 09_50_45 annotated](https://cloud.githubusercontent.com/assets/9324499/14551844/52e7121e-02a6-11e6-9655-15ea2f60e945.png)

**Comment by stefco** [comment]
invaluable:
http://wiki.bash-hackers.org/howto/calculate-dc

**Comment by stefco** [comment]
This was useful:

http://www.unix.com/shell-programming-and-scripting/80633-sed-combining-multiple-lines-into-one.html

``` bash
# bitrate of the irig-b signal
bitrate=16384

# what is considered a high vs. low signal; set it to 3500 so that
# ringing doesn't get read as too high
threshold=3500

# where within each character to test
t1=20
t2=60
t3=110
t4=150

# how much to add using dc, since dc will just add the difference between test points
a1=t1
a2=$(( t2 - t1 ))
a3=$(( t3 - t2 ))
a4=$(( t4 - t3 ))

# how each character type is represented; these are the only acceptable line values
rep_0="1000"
rep_1="1100"
rep_c="1110"

# make a temporary file to hold the raw output of the read
tempfile=$(mktemp)

# i think it is pretty clear what the next line does
sed -n "$(echo {0..99}" ${bitrate}*100/${a1}+p${a2}+p${a3}+p${a4}+p" | dc | sed 's/$/p/')" \
    | sed 's_$_ '"${threshold}"'/p_' \
    | dc \
    | sed 'N;N;N;s/\'$'\n''//g;s/'${rep_0}'/0/;s/'${rep_1}'/1/;s/'${rep_c}'/c/;/..../q1' \
    >| "${tempfile}"
```

**Comment by stefco** [comment]
How to convert from day of year to date, both using GNU `date` and POSIX `date`:

http://superuser.com/questions/231692/how-to-convert-from-day-of-year-and-year-to-a-date-yyyymmdd

**Comment by stefco** [comment]
read all lines of a file into an array:

http://stackoverflow.com/questions/11393817/bash-read-lines-in-file-into-an-array

**Comment by stefco** [comment]
better example of reading lines into an array (that hopefully actually works, and is portable): 

http://mywiki.wooledge.org/BashFAQ/005?highlight=%28readarray%29#Loading_lines_from_a_file_or_stream

**Comment by stefco** [comment]
lovely, it's working!

``` bash
$ geco_dump -s 'Nov 1 2015' -e 'Nov 1 00:01:00 2015' -c H1:CAL-PCALX_IRIGB_OUT_DQ -T | geco_irig_decode
Nov 01 00:00:17 2015
```

**Comment by stefco** [comment]
Implemented in 4052e68!

**Comment by stefco** [comment]
the help file:

```
geco_irig_decode -h
USAGE:

    <source_of_timeseries> | geco_irig_decode

    geco_irig_decode takes the first second worth of an IRIG-B timeseries from
    stdin and decodes the time it represents, then prints that time to stdout.
    You can pipe in as much data as you want, but it will only decode the first
    second worth of data. This helps guarantee its speed, even if the user is
    foolishly pumping gallons of data into the script.

    It will also check to make sure that the signal was not *too* erratic, that
    all control characters are in the right places within the signal, and that
    no illegal waveforms popped up. The script is implemented in pure bash and
    is adaptable. It is best used with geco_dump as the data source for the
    sake of convenience.

OPTIONS:

    -h                    print this help dialog.
```
